src/applications/toe/helpers.jsx

```diff
@@ -99,29 +99,72 @@ export function prefillTransformer(pages, formData, metadata, state) {
   const claimant = state.data?.formData?.data?.attributes?.claimant || {};
   const contactInfo = claimant?.contactInfo || {};
   const sponsors = state.data?.formData?.attributes?.sponsors;
+
+  const stateUser = state.user;
+  const vaProfile = stateUser?.vaProfile;
+  const profile = stateUser?.profile;
+  const vet360ContactInfo = stateUser.vet360ContactInformation;
+
+  const emailAddress =
+    profile?.email ||
+    vet360ContactInfo?.email?.emailAddress ||
+    contactInfo.emailAddress ||
+    undefined;
+
+  let mobilePhoneNumber;
+  let mobilePhoneIsInternational;
+  const v360mp = vet360ContactInfo?.mobilePhone;
+  if (v360mp?.areaCode && v360mp?.phoneNumber) {
+    mobilePhoneNumber = [v360mp.areaCode, v360mp.phoneNumber].join();
+    mobilePhoneIsInternational = v360mp.isInternational;
+  } else {
+    mobilePhoneNumber = contactInfo?.mobilePhoneNumber;
+  }
+
+  let homePhoneNumber;
+  let homePhoneIsInternational;
+  const v360hp = vet360ContactInfo?.homePhone;
+  if (v360hp?.areaCode && v360hp?.phoneNumber) {
+    homePhoneNumber = [v360hp.areaCode, v360hp.phoneNumber].join();
+    homePhoneIsInternational = v360hp.isInternational;
+  } else {
+    homePhoneNumber = contactInfo?.homePhoneNumber;
+  }
+
   const newData = {
     ...formData,
     sponsors,
     formId: state.data?.formData?.data?.id,
     claimantId: claimant.claimantId,
-    'view:userFullName': {
-      userFullName: {
-        first: claimant.firstName || undefined,
-        middle: claimant.middleName || undefined,
-        last: claimant.lastName || undefined,
+    [formFields.viewUserFullName]: {
+      [formFields.userFullName]: {
+        first:
```

Comment:
When I implemented this for MEB, I treated name as a block. So, just in case names are different, they wouldn't be mixed and matched.

If the VA profile is {givenNames: ["John"], familyName: "Smith} and the claimant call returns {firstName: "Jane", middleName: "Louise", lastName: "Doe"}, the result with your logic would be "John Louise Smith".

This is my implementation: https://github.com/department-of-veterans-affairs/vets-website/pull/22215/files#diff-61b93329fb83366b254dccccdcb3db442c05dc3325d498e1966822db818878c8R228